### PR TITLE
Update v8-jsi to 0.71.12 from 0.71.3

### PR DESCRIPTION
**Pull request by docEdub**

### CODE REVIEW

File: Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp [hunk 1]
```diff
@@ -4,47 +4,68 @@
...
         Run(env);
 
-        result = napi_ext_close_env_scope(env, scope);
+        result = jsr_close_napi_env_scope(_env, scope);
         assert(result == napi_status::napi_ok);
+
+        // TODO: jsr_delete_runtime needs to be called somewhere, but not here.
+        // See https://github.com/microsoft/v8-jsi/blob/559e188e3d13a97b75372119fde2b4a53a1addbc/src/testmain.cpp#L41
+        //result = jsr_delete_runtime(runtime);
+        //assert(result == napi_status::napi_ok);
```
Comment (docEdub) [inline Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp]: I'm not sure where this is supposed to go, so I left it as a TODO for when Sergio formalizes it all.

Comment (SergioRZMasson) [reply Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp]: Why can't we call jsr_delete_runtime here? As far as I know the Javascript thread will be looping in Run(env) until it is time to shutdown, in that case it should be correct to terminate the runtime.

Comment (docEdub) [reply Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp]: Ya, I thought so too and had it uncommented before but found that there is additional cleanup happening after this function returns that expects the JS env to still be alive.

Comment (SergioRZMasson) [reply Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp]: Where is additional cleanup been done?

Comment (docEdub) [reply Core/AppRuntime/Source/AppRuntime_Napi-c-external.cpp]: Discussed this w/ Sergio and got issue resolved.

**Pull request merged**